packages/replayio/src/utils/authentication/getAccessToken.ts

```diff
@@ -0,0 +1,72 @@
+import { readFromCache, writeToCache } from "../cache";
+import { updateLaunchDarklyCache } from "../launch-darkly/updateLaunchDarklyCache";
+import { maskString } from "../maskString";
+import { cachedAuthPath } from "./config";
+import { debug } from "./debug";
+import { refreshAccessTokenOrThrow } from "./refreshAccessTokenOrThrow";
+import { CachedAuthDetails } from "./types";
+
+export async function getAccessToken(): Promise<string | undefined> {
+  if (process.env.REPLAY_API_KEY) {
+    debug("Using token from env (REPLAY_API_KEY)");
+    return process.env.REPLAY_API_KEY;
+  } else if (process.env.RECORD_REPLAY_API_KEY) {
+    debug("Using token from env (RECORD_REPLAY_API_KEY)");
+    return process.env.RECORD_REPLAY_API_KEY;
+  }
+
+  let { accessToken, refreshToken } = readFromCache<CachedAuthDetails>(cachedAuthPath) ?? {};
+  if (typeof accessToken !== "string") {
+    debug("Unexpected accessToken value: " + accessToken);
+    return;
+  } else if (typeof refreshToken !== "string") {
```

Comment:
We still use `refreshToken` unconditionally later on in this function so we should check both here

```suggestion
  }
  if (typeof refreshToken !== "string") {
```